### PR TITLE
changes to load_dataset()

### PR DESCRIPTION
**Pull request by liameabbott**
@tpoterba Some changes to the load_dataset() function and documentation.

@jbloom22 I reworded the "Datasets" documentation page. Is it clearer now, or is there anything else you would like to add?

### CODE REVIEW

**Comment by jbloom22** [comment]
It's much clearer, awesome

File: hail/python/hail/experimental/datasets.py [hunk 1]
```diff
@@ -2,56 +2,63 @@
...
-        Path of the datasets configuration file.
-        Leave as default if running on Google Cloud Platform.
 
     Returns
     -------
     :class:`.Table` or :class:`.MatrixTable`"""
 
-    with hl.hadoop_open(config_file, 'r') as f:
-        config = json.load(f)
+    with hl.hadoop_open('gs://hail-datasets/datasets.json', 'r') as f:
```
Comment (tpoterba) [inline hail/python/hail/experimental/datasets.py]: why not parameterize this?

File: hail/python/hail/experimental/datasets.py [hunk 2]
```diff
@@ -2,56 +2,63 @@
...
+        datasets = json.load(f)
 
-    builds = [{'path': x['path'], 'reference_genome': x['reference_genome']}
-              for x in config if x['name'] == dataset_name]
+    names = set([dataset['name'] for dataset in datasets])
+    if name not in names:
+        raise ValueError('"{}" is not a dataset available in the repository.'.format(name))
```
Comment (tpoterba) [inline hail/python/hail/experimental/datasets.py]: calling `repr(name)` means you don't need to explicitly include quotes, and also means this is resilient to the user passing in something with newlines/tabs etc

**Comment by liameabbott** [comment]
@danking I can't really figure out what's going wrong with the build here -- any ideas?

**Comment by tpoterba** [comment]
sorry, my bad, will fix the artifacts

**Comment by tpoterba** [comment]
the docs build failed

**Pull request merged**